addons/stock/models/stock_move.py

```diff
@@ -1555,11 +1555,16 @@ def _update_reserved_quantity(self, need, available_quantity, location_id, lot_i
                 to_update.with_context(reserved_quant=reserved_quant).reserved_uom_qty += uom_quantity
             else:
                 if self.product_id.tracking == 'serial':
-                    self.env['stock.move.line'].with_context(reserved_quant=reserved_quant).create([self._prepare_move_line_vals(quantity=1, reserved_quant=reserved_quant) for i in range(int(quantity))])
+                    vals = self._add_serial_move_line_to_vals(reserved_quant, quantity)
+                    if vals:
+                        self.env['stock.move.line'].with_context(reserved_quant=reserved_quant).create(vals)
                 else:
                     self.env['stock.move.line'].with_context(reserved_quant=reserved_quant).create(self._prepare_move_line_vals(quantity=quantity, reserved_quant=reserved_quant))
         return taken_quantity
 
+    def _add_serial_move_line_to_vals(self, reserved_quant, quantity):
```

Comment:
```suggestion
    def _add_serial_move_line_to_vals_list(self, reserved_quant, quantity):
```

We don't really return a dict containing the vals, we return a list of vals. 🙂